ttkwidgets/tickscale.py

```diff
@@ -0,0 +1,633 @@
+            digits = max(self._digits, interv, resol, from_, to)
+            if digits != self._digits:
+                self._digits = digits
+                self._formatter = '{:.' + str(self._digits) + 'f}'
+                reinit = True
+        if 'variable' in kw:
+            self._var = kw['variable']
+            if not self._var:
+                self._var = tk.DoubleVar(self, self.get())
+                kw['variable'] = self._var
+            try:
+                self._var.trace_add('write', self._increment)
+            except Exception:
+                # backward compatibility
+                self._var.trace('w', self._increment)
+
+        self.scale.configure(**kw)
+
+        if 'from_' in kw or 'from' in kw or 'to' in kw:
+            self._extent = self.scale.cget('to') - self.scale.cget('from')
+            self._start = self.scale.cget('from')
+            reinit = True
+        if 'style' in kw:
+            self._style_name = kw['style']
+            if not self._style_name:
+                self._style_name = '%s.TScale' % (str(self.scale.cget('orient')).capitalize())
+            self._style_change()
+        if 'orient' in kw:
+            if kw['orient'] == 'vertical':
+                self._style_name = self._style_name.replace('Horizontal', 'Vertical')
+                if 'tickpos' not in kw:
+                    self._tickpos = 'w'
+            else:
+                self._style_name = self._style_name.replace('Vertical', 'Horizontal')
+                if 'tickpos' not in kw:
+                    self._tickpos = 's'
+            self.scale.configure(style=self._style_name)
+            reinit = True
+        if reinit:
+            self._init()
+
+    def config(self, cnf={}, **kw):
+        self.configure(cnf=cnf, **kw)
+
+    def get(self):
+        if self._digits >= 0:
+            return round(self.scale.get(), self._digits)
+        else:
+            return self.scale.get()
+
+    def convert_to_pixels(self, value):
+        """Convert value in the scale's unit into a position in pixels."""
+        percent = ((value - self._start) / self._extent)
+        return percent * (self.get_scale_length() - self._sliderlength) + self._sliderlength / 2
+
+    def _update_slider_length_horizontal(self):
+        """Measure the length of the slider and update the value of self._sliderlength."""
+        if not self.scale.identify(2, 2):
+            # to soon, wait for the scale to be properly displayed
+            # binding to <Map> event does not work, it can still be to soon to
+            # get any result from identify
+            self.after(10, self._update_slider_length_horizontal)
+        else:
+            w = self.scale.winfo_width()
+            i = 0
+            while i < w and 'slider' not in self.scale.identify(i, 2):
+                i += 1
+            j = i
+            while j < w and 'slider' in self.scale.identify(j, 2):
+                j += 1
+            if j - i == 0:
```

Comment:
I think this would be more readable with `j == i`, and it satisfies the same condition.

Reply:
Done.